Flowsheet_Class/flowsheet.py

```
@@ -168,19 +183,54 @@ def create_from_sfiles(self, sfiles_in="", overwrite_nx=False, merge_HI_nodes=Tr
                     # Branch opens. Looping through the branch until it is terminated.
                     elif self.sfiles_list[token_idx + step] == "[":
                         branches = 1
                         found = False
+                        branch_step = 0
                         while not (token_idx + step) == last_index:
                             step += 1
+                            branch_step += 1
                             if not found and bool(re.match(pattern_node, self.sfiles_list[token_idx + step])):
+                                if branch_step != 1:
+                                    raise AssertionError(
+                                        f"Error in create_from_sfiles: found = False but branch_step != 1." 
```

Ruff (0.15.9)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)